pkg/virt-controller/services/template.go

```diff
@@ -401,6 +406,38 @@ func getMemoryOverhead(domain v1.DomainSpec) *resource.Quantity {
 	return overhead
 }
 
+func getPortsFromVMI(vmi *v1.VirtualMachineInstance) []k8sv1.ContainerPort {
+	ports := make([]k8sv1.ContainerPort, 0)
+
+	configuredPorts := make(map[string]struct{})
+	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
+		if iface.Slirp != nil && iface.Slirp.Ports != nil {
+			for _, port := range iface.Slirp.Ports {
+				if port.Protocol == "" {
+					port.Protocol = "TCP"
```

Comment:
You could copy it over blindly too since we follow the same pattern like on the pod.

Reply:
No problem

Reply:
The copy will work but I need the protocol to be configured, I use it to create a unique name for the containerPort struct.